pkg/cloudprovider/vsphere/instances.go

```diff
@@ -165,6 +167,13 @@ func (i *instances) InstanceExistsByProviderID(ctx context.Context, providerID s
 		return true, nil
 	}
 
+	if err == vclib.ErrNoVMFound {
+		if _, ok := os.LookupEnv("ENABLE_DELETION_OF_VM"); ok {
```

Comment:
One more thing, should this be something in the cloud config instead of an env var?

Reply:
That's a good question... do we have a feeling what option most users are going to select? Currently, the original behavior is preserved for backward compatibility... if we feel that users going to all go one way, then maybe the environment variable is good enough for the time being. Another way to look at it is, do we think this option is going to be around for a very very long time, if so then we should have it in the config.

If we do have it in the config, then it should go here in a new struct called like CPIOptions or something:
https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphere/config/types_yaml.go#L45

Reply:
Talked about this and came to agreement that we should fall in-line with other cloud provider controller behaviors.